cmd/crypto/vault.go

```diff
@@ -130,70 +129,80 @@ func NewVault(config VaultConfig) (KMS, error) {
 	return v, nil
 }
 
-// reauthenticate() tries to login in 1 minute
-// intervals until successful.
-func (v *vaultService) reauthenticate() {
-	retryDelay := 1 * time.Minute
-	go func() {
-		for {
-			if err := v.authenticate(); err != nil {
-				time.Sleep(retryDelay)
-				continue
-			}
-			return
-		}
-	}()
-}
-
-// renewer calls vault client's renewer that automatically
-// renews secret periodically
-func (v *vaultService) renewer(secret *vault.Secret) {
+// renewSecret tries to renew the given secret. It blocks
+// until it receives either the new secret or encounters an error.
+func (v *vaultService) renewSecret(secret *vault.Secret) (*vault.Secret, error) {
 	renewer, err := v.client.NewRenewer(&vault.RenewerInput{
 		Secret: secret,
 	})
 	if err != nil {
 		logger.FatalIf(err, "crypto: hashicorp vault token renewer could not be started")
```

Comment:
@aead, just return the error here instead of logging this fatal message. Otherwise for every network down situation, the minio server will stop instead of retrying login to vault.

Reply:
We should remove this FatalIf message @aead

Reply:
Hm, I actually do not agree here. `NewRenewer` only returns an error if the the input argument is invalid. That means we have a bug in our code here. Please notice that renewal does not start by creating this struct... I rather suggest to change the log msg to `crypto: failed to create hashicorp fault token renewer`.

Reply:
Perhaps this should be `logger.CriticalIf` such that server can catch it